### PR TITLE
DLPX-71708 Recovery environment fails to boot

### PR DESCRIPTION
**Pull request by pcd1193182**
http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/3950/console

Manual testing performed with an ab-pre-push VM; machine was rebooted repeatedly until it entered the recovery environment, at which time ssh and http connectivity was verified. The package was then reinstalled, to emulate the upgrade workflow, and the test was repeated. All tests were successful.

### CODE REVIEW

**Comment by pzakha** [comment]
@pcd1193182 Could you add some context (and comments to the code) explaining what's going on? From what I'm understanding we were somehow using the wrong version of the linux kernel?

**Comment by grwilson** [comment]
To add to Pavel's comment, can you also update the bug too? I'm also missing the context.

**Comment by pcd1193182** [comment]
@pzakha @grwilson done and done

File: debian/postinst [hunk 1]
```diff
@@ -16,7 +17,20 @@ gunzip <"$base_img" | cpio -idm
...
 
-modpath="/lib/modules/$(uname -r)/"
+. /usr/lib/grub/grub-mkconfig_lib
+
+#
+# Determine the latest installed kernel version, and sync kernel modules for
+# that version into the recovery environment.
+#
+list=
+for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-*; do
```
Comment (pzakha) [inline debian/postinst:27]: We should look exclusively for `/boot/vmlinuz-*` here, given that this is where we are installing the file. If for some reason we get hits in other directories as well, `version_find_latest` will likely not work properly, and if we get a hit for `/boot/kernel-*` this line will break:
``` modpath="/lib/modules/${basename#vmlinuz-}" ```

**Pull request merged**